# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../linux26/gcc3.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

arch/x86/kernel/smpboot.c:750:70: unterminated argument list invoking macro "startup_ipi_hook"
arch/x86/kernel/smpboot.c: In function `wakeup_secondary_cpu':
arch/x86/kernel/smpboot.c:754: implicit declaration of function `startup_ipi_hook'
arch/x86/kernel/smpboot.c:754: syntax error before "unsigned"

gcc version 3.2.3 - Yuck.

  - Rene Rebe <rene@exactcode.de>

--- linux-2.6.26/arch/x86/kernel/smpboot.c.vanilla	2008-07-13 23:51:29.000000000 +0200
+++ linux-2.6.26/arch/x86/kernel/smpboot.c	2008-09-30 14:59:43.000000000 +0200
@@ -747,11 +747,10 @@
 	 * Paravirt / VMI wants a startup IPI hook here to set up the
 	 * target processor state.
 	 */
-	startup_ipi_hook(phys_apicid, (unsigned long) start_secondary,
 #ifdef CONFIG_X86_64
-			 (unsigned long)init_rsp);
+	startup_ipi_hook(phys_apicid, (unsigned long) start_secondary, (unsigned long)init_rsp);
 #else
-			 (unsigned long)stack_start.sp);
+	startup_ipi_hook(phys_apicid, (unsigned long) start_secondary, (unsigned long)stack_start.sp);
 #endif
 
 	/*
